modules/packages/ConcurrentMap.chpl

```diff
@@ -19,6 +19,12 @@
  */
 
 /*
+  .. note::
+
+    This module relies on the :mod:`AtomicObjects` package, which currently
+    requires ``extern`` code block support. See the documentation of the
+    :mod:`AtomicObjects` package for more details.
```

Comment:
I'd say more directly that it requires the compiler to be built with LLVM.  Put another way, I don't think we can/should assume that users will necessarily know what an extern block is or the implications it has on how the compiler was built.

Didn't the skipif that I commented on which led to this PR also list almost every C back-end compiler as not being supported because of some reliance on assembly or a gnu-ism or something?  If so, that needs to be said as well, I think?  (where ideally we'd phrase it as "only supported with these back-ends" rather than "not supported on all of these back-ends").

Reply:
This is due to reliance on the `AtomicObjects` module. Rather than restate everything that was already said there I wanted to just link to it. 

See: https://chapel-lang.org/docs/modules/packages/AtomicObjects.html

I can adjust the comment to also mention architecture restrictions.

Reply:
> Rather than restate everything that was already said there I wanted to just link to it.

I mean, in the limit, you could remove the bit about `extern` blocks here as well, and AtomicObjects could remove the part about LLVM being required for extern blocks, requiring the user to trace all the way through to the documentation about extern blocks to understand why their compile is breaking, but that doesn't seem like it would make anyone very happy.

If we were opting to optimize for avoiding copy-pasting or verbose text, I'd simply say:

```
This module currently relies on the Chapel compiler to be built with LLVM support.  [and whatever we need to say about target C compilers]
```

since that's the only thing an end-user is likely to care about ("What do I need to do to use this?").  It's not obvious to me that a client of the code cares much about what other modules it uses or why the requirement is there, which feel more like implementation details and obscure language interop rules.  And a developer who cares about those things can likely guess or find out by seeing what happens if they violate the constraint.

Reply:
How about:

```
This module relies on the AtomicObjects package, which currently requires that the Chapel compiler be built with LLVM support, and restricts the target platform to x86_64.

See the documentation of the AtomicObjects package for details.
```

Reply:
Doesn't the skipif also say that GNU must be used?

```
# AtomicObjects uses GCC inline assembly, so only test it on gcc/clang
```

(and why do I feel like I'm suddenly the expert on this, simply from reading the skipif?)

Reply:
My concern is more of that I don't really want to enumerate all the restrictions in every module that uses `AtomicObjects`, because there are at least 3 of those. I would personally rather just hint that there are restrictions.

Reply:
I'm trying to put myself in the shoes of a user:  Somehow I find myself randomly on the ConcurrentMap page (because I usually randomly find my way to a library's documentation through searching or someone pointing me to it, not by reading through all of the documentation in alphabetical order).  I use the module, compile my code and... things blow up, maybe with a nice error message and maybe not.  To not have those portability limitations noted in an in-your-face way in the module documentation seems like an exercise in frustration for someone who always uses the CCE compiler on arm as their back-end, say.  Or, maybe I _am_ using gcc on x86, but am trying to build something cool and portable using ConcurrentMap for people to consume on other platforms, yet don't realize I'm depending on a non-portable module because the docs didn't tell me it wasn't and my compile worked fine.

I'm not trying to pick on you or this module per se, but am sensitized to this because you asked me to help review the docs the other week, and in doing so, I wasn't aware of these limitations (because the documentation doesn't state them).  And maybe you weren't either, since we didn't catch that the skipif was necessary until the testing ran—so if it had been documented, maybe it would've helped us catch it in code review?

Copy-pasting a portability warning into 3 files doesn't seem like a big deal to me compared to all the other work we do on a daily basis (what I wouldn't give to trade you the CHANGES file editing for adding them!), and keeping users from stepping into holes seems worth the level of effort it seems like it should take.

Or you could take the intentionally lazy approach and just put it into this one file since it's the one you took under your wing, and the one which will be highlighted in 1.25.1 potentially leading people to it.  But then we should open an issue to note that the other two also need updating, and that seems like almost as much work as simply doing it.

Clearly, the "right" solution here would be to get someone to make these underlying modules more portable, but I'm not asking anyone to do that.

Reply:
Ok, I can update the `AtomicObjects` module to mention its reliance on `GCC` style inline assembly, and then I can more-or-less copy and paste the restrictions into all three modules which rely on `AtomicObjects`.

Reply:
Thanks!

Reply:
Ok, I've gone ahead and edited/copied the warning.